src/t8_forest/t8_forest.c

```diff
@@ -369,19 +369,69 @@ t8_forest_comm_global_num_elements (t8_forest_t forest)
 static int
 t8_forest_refine_everything (t8_forest_t forest, t8_forest_t forest_from,
                              t8_locidx_t which_tree, t8_locidx_t lelement_id,
-                             t8_eclass_scheme_c * ts, const int is_family,
-                             const int num_elements,
-                             t8_element_t * elements[])
+                             t8_eclass_scheme_c *ts, const int is_family,
+                             const int num_elements, t8_element_t *elements[])
 {
 
   return 1;
 }
 
+/**
+ * Check if any tree in a forest refines irregularly
+ * 
+ * \param[in] forest    The forest to check
+ * \return int          non-zero if any tree refines irregular
+ */
+static int
+t8_forest_refines_irregular (t8_forest_t forest)
+{
+  int                 irregular = 0;
+  int                 int_eclass;
+  for (int_eclass = (int) T8_ECLASS_ZERO; int_eclass < (int) T8_ECLASS_COUNT;
+       int_eclass++) {
+    if (forest->cmesh->num_local_trees_per_eclass[int_eclass] > 0) {
+      irregular = irregular
+        || t8_eclass_refines_irregular ((t8_eclass_t) int_eclass);
```

Comment:
See comment on the refines_irregular function.

The call must depend on the eclass_scheme not on the eclass itself.

A single eclass can have one scheme implementation that refines "regular" and one that does not.

Reply:
Done